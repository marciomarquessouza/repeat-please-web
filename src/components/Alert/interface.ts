export type TVisbility = 'visible' | 'hidden' | null;

export type TAlertTypes = 'error' | 'warning' | 'success';

export interface IAlertBoxProps {
	visibility: TVisbility;
}

export interface IAlertProps {
	visibility: TVisbility;
	message: string;
	type: TAlertTypes;
	onClose: () => void;
}

export interface IAlertTitleProps {
	type: TAlertTypes;
}
