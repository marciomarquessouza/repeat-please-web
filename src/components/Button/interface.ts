export interface IButtonProps {
	children: string;
	onClick: () => void;
	variant: 'primary' | 'secondary';
	circular?: boolean;
}
