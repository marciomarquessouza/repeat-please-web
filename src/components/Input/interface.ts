export interface IInputProps {
	type: 'email' | 'password' | 'text' | 'number';
	id?: string;
	name?: string;
	autoComplete?: 'on' | 'off';
	autoFocus?: boolean;
	placeholder?: string;
	value: string;
	onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}
