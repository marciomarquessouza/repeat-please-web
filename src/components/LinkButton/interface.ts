export interface ILinkProps {
	onClick: () => void;
	children: string;
}
