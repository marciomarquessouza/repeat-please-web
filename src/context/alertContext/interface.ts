import { TAlertTypes, TVisbility } from '../../components/Alert/interface';

export interface IAlertProviderProps {
	children: React.ReactNode;
}

export interface IAlertContext {
	showAlert: (message: string, type: TAlertTypes) => void;
}

export interface IAlertState {
	visibility: TVisbility | null;
	message: string;
	type: TAlertTypes;
}
