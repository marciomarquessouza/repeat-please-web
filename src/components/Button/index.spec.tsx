import React from 'react';
import { shallow } from 'enzyme';
import { Button } from '.';
import { IButtonProps } from './interface';
import { findByDataTest } from '../../utils/tests/findByDataTest';

const defaultProps: IButtonProps = {
	children: 'Button',
	onClick: jest.fn(),
	variant: 'primary',
};

const setup = (props: IButtonProps) =>
	shallow(<Button {...props}>{props.children}</Button>);

describe('#ButtonSquare', () => {
	it('renders properly', () => {
		const wrapper = setup(defaultProps);
		expect(wrapper).toMatchSnapshot();
	});

	describe('when the button is pressed', () => {
		it('calls onClick function', () => {
			const wrapper = setup(defaultProps);
			const button = findByDataTest(wrapper, 'button-square');
			button.simulate('click');
			expect(defaultProps.onClick).toHaveBeenCalled();
		});
	});
});
