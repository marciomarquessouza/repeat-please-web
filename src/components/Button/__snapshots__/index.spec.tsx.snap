// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ButtonSquare renders properly 1`] = `
<styled.a
  data-test="button-square"
  onClick={[MockFunction]}
  variant="primary"
>
  Button
</styled.a>
`;
